src/main/java/LacksDataBaseProject/SetLack.java

```diff
@@ -0,0 +1,41 @@
+package LacksDataBaseProject;
+
+import LacksDataBaseProject.Exceptions.LessThanZeroException;
+import LacksDataBaseProject.Exceptions.PurchaserAccessException;
+import LacksDataBaseProject.Exceptions.WrongDateFormatException;
+
+public class SetLack implements CheckUserRole {
+
+    @Override
+    public boolean checkUserRole(User user) throws PurchaserAccessException {
+        if (user.getRole() == Role.PURCHASER) {
+            return true;
+        } else
+            throw new PurchaserAccessException("No authorization. Only Purhcaser allowed to modify this data. ");
+    }
+
+    protected void setDate(Lack lack, User user) throws WrongDateFormatException, PurchaserAccessException {
+        if (checkUserRole(user)) {
+            lack.setExpectedDeliveryDateAndTime(App.getExpectedDeliveryDateAndTime());
+            System.out.println("Sending Alert to " + lack.getForwarderSkypeID() + ". "
```

Comment:
Tutaj też proponuje się pozbyć zmiennych stałych :) jeśli chcecie coś logować to proponuje wykorzystać jakiś logger, np: https://github.com/JakeWharton/timber . Dajcie znać czy mieliście coś o loggerach, bo jeśli nie podeślę więcej info :)

Reply:
ok